### PR TITLE
Add Truffleruby to CI

### PR DESCRIPTION
**Pull request by gogainda**

### CODE REVIEW

**Comment by gogainda** [comment]
@halostatue pls review/merge

File: .travis.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@ rvm:
   - jruby-9.1.17.0
   - jruby-head
   - ruby-head
+  - truffleruby-head
```
Comment (halostatue) [inline .travis.yml:14]: This should also be added to `matrix.allow_failures`. If there’s a major release version, we can also add that without `allow_failures`.

**Comment by gogainda** [comment]
@halostatue taken care of

**Comment by gogainda** [comment]
@eregon

File: .travis.yml [hunk 2]
```diff
@@ -11,10 +11,14 @@ rvm:
...
+  - truffleruby
 matrix:
   allow_failures:
     - rvm: jruby-head
     - rvm: ruby-head
+    - rvm: truffleruby-head
+    - rvm: truffleruby
```
Comment (eregon) [inline .travis.yml:21]: This one should probably not be in `allow_failures` from the other comments.

**Pull request merged**